### ISSUE TITLE
CutMix: Regularization Strategy to Train Strong Classifiers with Localizable Features

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#141]
**한 문장으로 요약하자면,
이미지의 일부분을 비율을 정해 0으로 cut & 그 부분을 랜덤으로 선택된 다른 이미지로 mix해 학습시키는 것.**

## Abstract & Introduction
* propose the CutMix augmentation strategy
* patches are cut and pasted among training images where the ground truth labels are also mixed proportionally to the area of the patches
* to prevent a CNN from focusing too much on a small set of intermediate activations or on a small region on input images
* --> random feature removal regularizations have been propose (cutout 같은 방법)
* Instead of simply removing pixels, we replace the removed regions with a patch from another image
* ImageNet classification/ImageNet localization/Pascal VOC 07 detection 테스크에 대한 Mixup, Cutout, CutMix 결과 비교.
  <image src="https://user-images.githubusercontent.com/67627471/132949562-a8f99354-ca67-4512-ba6b-36c9c14cbdf3.png" width="40%" height="40%" />

## 관련 연구
* Regional dropout
* Synthesizing training data
* Mixup
* Tricks for training deep networks

## CutMix 설명
* original 손실 함수로 train -> 식으로 표현.
  <image src="https://user-images.githubusercontent.com/67627471/132949698-9671520c-4760-490f-ba36-ef0c9aafeff0.png" width="25%" height="25%" />
* 정사각형 모양의 마스크 M 씌움. (그 부분만큼 날리는 것)
  <image src="https://user-images.githubusercontent.com/67627471/132949737-187ac589-ee75-4be7-b5e3-beed646fc686.png" width="30%" height="30%" />
  * 크롭된 영역 비율이 다음과 같도록 정함.
    <image src="https://user-images.githubusercontent.com/67627471/132949861-6dfff650-2bb5-4075-8208-b64cd87cb43e.png" width="15%" height="15%" />
  * 마스크 부분은 0으로 채움.
* 바닐라 ResNet-50 모델을 사용하여 여러 가지 증강 기술을 CAM으로 visualize해서 봄.
  <image src="https://user-images.githubusercontent.com/67627471/132949999-4b6eb83a-e0f4-4d5d-966f-fbcf726a14f4.png" width="30%" height="30%" />
* Mixup은 전체 이미지 영역을 사용하고 image와 라벨을 믹스.
* Cutout은 부분적인 dropout을 씀.
* CutMix는 위의 두 기술을 융합함. 그래서 아래에 다 해당함.
  * Usage of full image region
  * Regional dropout
  * Mixed image & label 

## 결과
* ResNet-50을 베이스로 한 image classification에서 CutMix가 가장 낮은 에러율을 보임.
  <image src="https://user-images.githubusercontent.com/67627471/132950237-d2169d30-063e-4909-a5aa-e404f284250b.png" width="35%" height="35%" />
* CIFAR-100에 regularization 방법을 적용해 그 에러율을 비교. CutMix+ShakeDrop이 가장 우수한 성능을 보여줌.
  <image src="https://user-images.githubusercontent.com/67627471/132950191-2e7a0d84-1a17-4557-b36b-1dd9cae4e316.png" width="40%" height="40%" />

##
* 왜 Cutout보다 CutMix의 성능이 더 좋을까?? 특정 영역이 아예 다른 이미지로 바뀌어 버리면 오히려 클래스의 특징을 학습함에 있어 혼란을 주지 않나....??

**Comment by ghuijo** [comment]
네이버 클로바 AI 팀에서 낸 논문.
